base/bootstrap.go

```diff
@@ -70,14 +70,12 @@ const (
 )
 
 type cachedBucket struct {
-	bucket     *gocb.Bucket
-	teardownFn func()
+	bucket      *gocb.Bucket
+	teardownFn  func()
+	refcount    int
+	shouldClose bool
```

Comment:
Can you add a comment here to describe the interaction between shouldClose, refcount and teardownFn()?  

It takes a bit of time to work through the handling just by reading the code, a summary here would be helpful in future.